demisto_sdk/commands/common/hook_validations/id.py

```diff
@@ -507,7 +507,7 @@ def get_commands_to_integration(self, file_name, file_path):
     @error_codes('PB110,PB117')
     def is_entity_version_match_playbook_version(self, implemented_entity_list_from_playbook,
```

Comment:
there are some logic issues here as discussed in person,
lets refactor this method. Broadly - group entities by id so we will have have a list of versions by id, and only compare to the min version